EmployeeList/Services/ImageService.swift

```diff
@@ -0,0 +1,67 @@
+//
+//  ImageService.swift
+//  EmployeeList
+//
+//  Created by Wang Siman on 1/29/21.
+//  Copyright © 2021 None. All rights reserved.
+//
+
+import Foundation
+import Combine
+import UIKit
+
+protocol ImageServiceType: APIServiceType{
+    func existsInDisk(_ key: String) -> Bool
+    func loadFromDisk(forKey key: String) -> AnyPublisher<Data, Error>
+    func saveToDisk(_ image: UIImage, forKey key: String) -> AnyPublisher<Bool, Error>
+}
+
+final class ImageService: ImageServiceType {
+    internal let baseURL: String
+    internal let session: URLSession = URLSession.shared
+    internal let bgQueue: DispatchQueue = DispatchQueue.main
```

Comment:
You've called this `bgQueue`, which makes me think it's in the background, but it's the main queue.

Reply:
I grabbed the boilerplate from one of sample projects related to APIServiceType protocol. It defines bgQueue of DispatchQueue type. I guess this allows flexibility to configure if a service runs in the main thread or in background.

Reply:
I would just call it `queue`, not `bgQueue`, as `bg` implies background (or me :-)